test/integration/workers/publishing_api_document_republishing_worker_integration_test.rb

```diff
@@ -130,6 +130,18 @@ class PublishingApiDocumentRepublishingWorkerIntegrationTest < ActiveSupport::Te
 
       assert_all_requested(requests)
     end
+
+    test "Should not retry if encountering `GdsApi::HTTPPayloadTooLarge` exception" do
+      edition = create(:published_publication)
+      Whitehall::PublishingApi.expects(:publish).raises(GdsApi::HTTPPayloadTooLarge.new("413 Request Entity Too Large"))
+      worker = PublishingApiDocumentRepublishingWorker.new
+
+      assert_equal(worker.sidekiq_options_hash["retry"], true)
+      assert_raises(GdsApi::HTTPPayloadTooLarge) do
+        worker.perform(edition.document.id)
+        assert_equal(worker.sidekiq_options_hash["retry"], 0)
```

Comment:
Maybe this needs to come out of the do block? the worker.perform will raise an error which will short circuit this block. and the assert_equal never gets called. For example, i've added `assert_false true` to line 142 and the test was still passing

Reply:
Oof, good spot! I'm usually good at double-checking my test fails if I change the assertion. Fixed!